pyblish_maya/lib.py

```diff
@@ -313,3 +310,28 @@ def _show_no_gui():
 
     messagebox.setStandardButtons(messagebox.Ok)
     messagebox.exec_()
+
+
+class Dock(QtWidgets.QWidget):
+
+    def __init__(self, dock_count, parent=None):
+        super(Dock, self).__init__(parent)
+        QtWidgets.QVBoxLayout(self)
+        object_name = "pyblish_maya.dock" + str(dock_count)
+        self.setObjectName(object_name)
+
+
+def dock(window):
+
+    main_window = None
+    for obj in QtWidgets.qApp.topLevelWidgets():
+        if obj.objectName() == "MayaWindow":
+            main_window = obj
```

Comment:
I know it's unlikely, but this can produce a `main_window` variable that is still `None`. Either check that you got what you expect, or try a different approach.